### PR TITLE
docs: add TypeScript example for custom-queries.js

### LINKED ISSUE #889: TS example of custom-queries.js
### Describe the feature you'd like:

The current [setup documentation](https://testing-library.com/docs/react-testing-library/setup/) has a TS example for the `test-utils.js` script (namely `test-utils.tsx`). It does not, however, have a TS example of the `custom-queries.js` script. The ask is to update the documentation with such an example — I've tried implementing one myself with some difficulty.

### Suggested implementation:

Update the current document with an example as indicated above.

Example:

```ts
import { queryHelpers, buildQueries } from '@testing-library/react'

const queryAllByDataCy = (...args: unknown[]) =>
  queryHelpers.queryAllByAttribute('data-cy', ...args)

const getMultipleError = (c: HTMLElement, dataCyValue: unknown) =>
  `Found multiple elements with the data-cy attribute of: ${dataCyValue}`
const getMissingError = (c: HTMLElement, dataCyValue: unknown) =>
  `Unable to find an element with the data-cy attribute of: ${dataCyValue}`

const [
  queryByDataCy,
  getAllByDataCy,
  getByDataCy,
  findAllByDataCy,
  findByDataCy,
] = buildQueries(queryAllByDataCy, getMultipleError, getMissingError)

export {
  queryByDataCy,
  queryAllByDataCy,
  getByDataCy,
  getAllByDataCy,
  findAllByDataCy,
  findByDataCy,
}
```

Still seeing the following warning (not feeling comfortable with `unknown[]` but it looks like the underlying type is `Arguments extends any[]` and not sure how to correctly apply that here)

image

### Describe alternatives you've considered:

NA

### Teachability, Documentation, Adoption, Migration Strategy:

NA

### PR DESCRIPTION
**Pull request by nickmccurdy**
Fix #889

### CODE REVIEW

File: docs/react-testing-library/setup.mdx [hunk 1]
```diff
@@ -187,6 +192,54 @@ export {
 }
 ```
 
+  </TabItem>
+
+  <TabItem value="ts">
+
+```js title="custom-queries.ts"
```
Comment (MatanBobi) [inline docs/react-testing-library/setup.mdx:199]: I think the snippet type should be `ts` :)

File: docs/react-testing-library/setup.mdx [hunk 2]
```diff
@@ -155,6 +155,11 @@ elements by `data-cy`, a "test ID" convention mentioned in the
 [Cypress.io](https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements)
 documentation.
 
+  <Tabs groupId="test-utils" defaultValue="js" values={[ {label: 'Javascript',
+  value: 'js'}, {label: 'Typescript', value: 'ts'}, ]}>
```
Comment (MatanBobi) [inline docs/react-testing-library/setup.mdx:159]: [Nit]
Maybe we should write `Javascript` and `Typescript` with a capital `s`?
(Not sure we're following this rule in other places also, so this really isn't important)

**Review by MatanBobi — APPROVED** [review]
Looks good to me, sorry for the nit comments.

**Comment by MatanBobi** [comment]
@nickmccurdy Can we merge this? :)

File: docs/react-testing-library/setup.mdx (hunk 2 above)
Comment (eps1lon) [inline docs/react-testing-library/setup.mdx:159]:
```suggestion
  value: 'js'}, {label: 'TypeScript', value: 'ts'}, ]}>
```

File: docs/react-testing-library/setup.mdx [hunk 3]
```diff
@@ -155,6 +155,11 @@ elements by `data-cy`, a "test ID" convention mentioned in the
 [Cypress.io](https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements)
 documentation.
 
+  <Tabs groupId="test-utils" defaultValue="js" values={[ {label: 'Javascript',
```
Comment (eps1lon) [inline docs/react-testing-library/setup.mdx:158]:
```suggestion
  <Tabs groupId="test-utils" defaultValue="js" values={[ {label: 'JavaScript',
```

**Pull request merged**